x-pack/elastic-agent/pkg/agent/cmd/install.go

```diff
@@ -141,30 +145,33 @@ func installCmd(streams *cli.IOStreams, cmd *cobra.Command, args []string) error
 			}
 		}
 	}
-	cfgFile := paths.ConfigFile()
-	err = install.Install(cfgFile)
-	if err != nil {
-		return err
-	}
 
-	defer func() {
-		if err != nil {
-			install.Uninstall(cfgFile)
-		}
-	}()
-
-	if !delayEnroll {
-		err = install.StartService()
+	cfgFile := paths.ConfigFile()
+	if status != install.PackageInstall {
+		err = install.Install(cfgFile) // can't run install for package installed agent, but must stop the running service and re-enroll, then start the service?
```

Comment:
not sure if we need to do the commented steps, currently the enroll command will be ran with the `--from-install` flag, but I don't know if that's correct

Reply:
Not sure either, what is the behavior in the OS?

Reply:
Do not need to stop the running service, it should just enroll and then trigger a restart through the Elastic Agent control protocol. It should not need to be restarted through the service manager of the host.